test/customscript.spec.ts

```diff
@@ -85,110 +67,88 @@ describe('CustomScriptIdentity', () => {
+    const template = JSON.stringify(synthAssetArtifact);
+    const random = makeRandomCustomScriptIdentity(template);
+    expect(random.contract.namespace).toBe(TEST_NAMESPACE);
+    expect(random.contract.template).toBeDefined();
+
+    const addr = await random.getNextAddress({
+      borrowAsset: '2c5dfb37a33fe2acf5c5412b1ddbd58f6f3353578904f3ede0173b2867362463',
+      borrowAmount: 1000_00000000,
+      collateralAsset: '5ac9f65c0efcc4775e0baec4ec03abdde22473cd3cf33c0419ca290e0751b225',
+      collateralAmount: 1_50000000,
+      payoutAmount: 1500000,
+      oraclePk: '0x0000000000000000000000000000000000000000000000000000000000000000',
+      issuerPk: '0x0000000000000000000000000000000000000000000000000000000000000000',
+      issuerScriptProgram: '0x0000000000000000000000000000000000000000000000000000000000000000',
+      priceLevel: numberToString(20000),
+      setupTimestamp: numberToString(1656686483),
+    });
 
-    pset = Psbt.fromBase64(pset)
-      .updateInput(0, {
-        tapLeafScript: [
-          {
-            leafVersion: 0,
-            script: Buffer.from(leafToSpendScript!, 'hex'),
-            controlBlock: Buffer.alloc(33),
-          },
-        ],
-      })
-      .toBase64();
-
-    const signed = await blindAndSignPset(
-      pset,
-      utxos,
-      [id],
-      [addr.confidentialAddress],
-      [addr.confidentialAddress],
-      true
+    expect(addr.confidentialAddress).toBeDefined();
+    expect(addr.blindingPrivateKey).toBeDefined();
+    expect(addr.derivationPath).toBeDefined();
+    expect(addr.taprootHashTree).toBeDefined();
+    expect(addr.taprootInternalKey).toBe(
+      '0000000000000000000000000000000000000000000000000000000000000000'
     );
-    const txid = await broadcastTx(signed);
-    expect(txid).toBeDefined();
+    expect(addr.publicKey).toBeDefined();
+    expect(addr.contract).toBeDefined();
   });
 
-  test('should be able to auto-sign CHECKSIG tapscript owned by marina', async () => {
-    const covenantLeaf = `$${TEST_NAMESPACE} OP_CHECKSIG`;
-    const template = `eltr(c6047f9441ed7d6d3045406e95c07cd85c778e4b8cef3ca7abac09b95c709ee5, { asm(${covenantLeaf}), asm(OP_FALSE) })`;
-    const id = makeRandomCustomScriptIdentity(template);
-    const addr = await id.getNextAddress();
-
-    await faucet(addr.confidentialAddress, 10000);
-    const utxos = await getUnspents(addr);
-    expect(utxos.length).toBeGreaterThanOrEqual(1);
-
-    const { pset } = await createSendPset(
-      [
-        {
-          address: addr.confidentialAddress,
-          asset: networks.regtest.assetHash,
-          value: 1000,
-        },
-      ],
-      utxos,
-      networks.regtest.assetHash,
-      () => addr.confidentialAddress,
-      'regtest'
-    );
+  test('should be able to instatiate a contract identity, fund the contract and spend those funds', async () => {
+    const template = JSON.stringify(transferWithCaptchaArtifact);
+    const random = makeRandomCustomScriptIdentity(template);
```

Comment:
where you passing the constructor arguments? ie. the sum  (3+4=7?) and the pubkey?